### ISSUE TITLE
Improvement proposal to build native images

### ISSUE THREAD

**Issue by bufferings** [micronaut-projects/micronaut-profiles#137]
How about having these files to build native image to make it simpler & easier?

## Basic Information

I'm using Micronaut `1.1.0` and generated app with following command:

```sh
$ mn create-app hello-graal --features graal-native-image
```

## Build without Docker

build-native-image.sh

```sh
#!/bin/sh
./gradlew assemble
native-image --no-server -cp build/libs/hello-graal-*.jar
```

Then we just run this script to build native image:

```sh
$ ./build-native-image.sh
$ ./hello-graal
```

## Build with Docker

docker-build.sh

```sh
#!/bin/sh
./gradlew assemble # <-- added this
docker build . -t hello-graal
echo
echo
echo "To run the docker container execute:"
echo "    $ docker run -p 8080:8080 hello-graal" # <-- modified this
```

Dockerfile

```Dockerfile
FROM oracle/graalvm-ce:1.0.0-rc15 as graalvm
WORKDIR /work
COPY ./build/libs/hello-graal-*.jar .
RUN native-image --no-server -cp hello-graal-*.jar

FROM frolvlad/alpine-glibc
EXPOSE 8080
WORKDIR /app
COPY --from=graalvm /work/hello-graal .
ENTRYPOINT ["/app/hello-graal"]
```

Then we just run the script to build the docker image:

```sh
$ ./docker-build.sh
$ docker run -p 8080:8080 hello-graal
```

### build-native-image.sh

```sh
#!/bin/sh
./gradlew assemble
native-image --no-server -cp build/libs/hello-graal-*.jar
```

Until the previous version, we had `native-image-build.sh` but currently we don't. I checked the commit logs and understood that it was lost on the way of improving Dockerfile.

It would be better to have it back, but it should be very simple, because it became so simple to build native images thanks to your great effort for 1.1.0.

### docker-build.sh

```sh
#!/bin/sh
./gradlew assemble # <-- added this
docker build . -t hello-graal
echo
echo
echo "To run the docker container execute:"
echo "    $ docker run -p 8080:8080 hello-graal" # <-- modified this
```

On the way of improving the Dockerfile, a process to build JAR file is lost. So I added `./gradlew assemble` at the beginning.

In addition, I changed the comment in the last line to use publish option instead of host mode, because the host network mode is only for Linux.

https://docs.docker.com/network/host/

### Dockerfile

```Dockerfile
FROM oracle/graalvm-ce:1.0.0-rc15 as graalvm
WORKDIR /work
COPY ./build/libs/hello-graal-*.jar .
RUN native-image --no-server -cp hello-graal-*.jar

FROM frolvlad/alpine-glibc
EXPOSE 8080
WORKDIR /app
COPY --from=graalvm /work/hello-graal .
ENTRYPOINT ["/app/hello-graal"]
```

In the first stage, we only need JAR file to build native image. We don't need the source code. So I only copy the JAR file.

In the second stage, we only need native image generated by the first stage. So I only copy the native image.

**Comment by graemerocher** [comment]
We removed these files because as you said it is much simpler now to build a native image and the scripts only work on Unix systems and didn't seem worth the effort to maintain.

The Cloud Run example actually has a multi stage build file that produces an optimized image. See 
https://github.com/micronaut-projects/micronaut-gcp/blob/master/examples/hello-world-cloud-run-graal/Dockerfile

**Comment by bufferings** [comment]
I understood and agree with you. It's simple enough, so no problem to remove shell scripts. One small thing, but it might be better to modify the document a little bit, because I was confused when I followed the document that it says:

https://docs.micronaut.io/1.1.0/guide/index.html#graalServices

> To build your native image using Docker simply run:
> $ docker build . -t hello-world
> $ docker run --network=host hello-world

> Or use the provided script:
> $ ./docker-build.sh

But it failed because I needed to build JAR file before that. In addition, I noticed the host mode doesn't work except for Linux.

The Cloud Run Dockerfile example is nice!

I will continue to learn Micronaut, which is great :)
